aiida_nanotech_empa/utils/gaussian_spin_wc_postprocess.py

```diff
@@ -0,0 +1,190 @@
+import numpy as np
+
+from PIL import Image
+
+import matplotlib.pyplot as plt
+
+
+def print_orb_energies(prop_dict, n_orb=4):
+
+    nspin = len(prop_dict['homos'])
+    mo_e = np.array(prop_dict['moenergies'])
+
+    if nspin == 1:
+
+        homo_s0 = prop_dict['homos'][0]
+
+        i_start = homo_s0 - n_orb
+        i_end = homo_s0 + n_orb
+
+        print("{:>10} {:>16} {:>6}".format("i_orb", "E (eV)", "occ"))
+
+        for i_orb in range(i_end, i_start, -1):
+            occ = 2 if i_orb <= homo_s0 else 0
+            print("{:>10} {:16.4f} {:6}".format(i_orb, mo_e[0][i_orb], occ))
+
+        print("GAP: {:.4f} eV".format(mo_e[0][homo_s0 + 1] - mo_e[0][homo_s0]))
```

Comment:
Instead of printing things, can we return a dictionary instead?

This way we decouple parsing and displaying.

Reply:
The same goes for the other functions in this file.

Reply:
In principle, this module is just meant for generating a compact and well-formatted report. The small amount of parsing is quite tightly linked together with the printing, so i wouldn't separate these into two different modules or functions. I understand that it might be useful to be able to print this report in different contexts, but I would add that functionality in the future when the need arises in a separate PR.

Reply:
My vision on this is that data generation/data parsing/data visualisation should be decoupled. The functions in this module provide a nice mechanism to extract relevant data. However, the visualisation should probably to go the notebooks.